### PR TITLE
Generating correct pom.xml when spring-d-m is used

### PR DESCRIPTION
**Pull request by TomasHofman**
"WebAuthn4J gradle build in PNC does not generate pom.xml files with versions of used dependencies"

NCL-5208

Fixed issues:
* Analyzer: dependencies that inherit version from BOM were missing version in generated manipulation.json keys.
  E.g., manipulation.json would contain:
  ```
      "alignedDependencies" : {
        "ch.qos.logback:logback-classic" : {
          "groupId" : "ch.qos.logback",
          "artifactId" : "logback-classic",
          "version" : "1.2.3.redhat-00001"
        }
      }
  ```
  while it should have been:
  ```
  "alignedDependencies" : {
        "ch.qos.logback:logback-classic:1.2.3" : {
          "groupId" : "ch.qos.logback",
          "artifactId" : "logback-classic",
          "version" : "1.2.3.redhat-00001"
        }
      }
  ```
* Manipulator:
    * Don't prevent spring-d-m from modifying the pom.xml - thus spring-d-m can include BOM references in pom.xml, which resolves the problem with missing dependency versions.
    * Fix version resolution when spring-d-m is used in combination with maven-publish plugin.

### CODE REVIEW

**Comment by TomasHofman** [comment]
Please do not merge this yet, I'm gonna test on more real life projects, but I wanted to make this visible. So far I was only testing on webauthn4j.

Interestingly I didn't notice any negative behavior when spring-d-m is allowed to modify the pom.xml. I didn't see it rewrite any of our changes, it just adds managed dependency section with imported BOMs.

**Comment by rnc** [comment]
Thats great!! Thanks very much for this :-)

**Comment by geoand** [comment]
Awesome!

**Comment by TomasHofman** [comment]
I thinks it's ready for review.

Note that I'm not manipulating versions in <managedDependencies> section, only in direct dependencies. So the managed dep will have original version, but that will be effectively rewritten by direct dep which would have updated version.

**Comment by geoand** [comment]
Neat!!!

File: manipulation/src/functTest/java/org/jboss/gm/manipulation/SimpleProjectWithMisconfiguredSpringDMAndMavenPluginsFunctionalTest.java [hunk 1]
```diff
@@ -23,6 +24,7 @@
     public TemporaryFolder tempDir = new TemporaryFolder();
 
     @Test
+    @Ignore("Disabling generatedPomCustomization is no longer necessary.")
     public void ensureProperPomGenerated() throws IOException, URISyntaxException {
```
Comment (rnc) [inline manipulation/src/functTest/java/org/jboss/gm/manipulation/SimpleProjectWithMisconfiguredSpringDMAndMavenPluginsFunctionalTest.java]: If this test is no longer applicable why not just remove it ?

File: manipulation/src/functTest/java/org/jboss/gm/manipulation/SimpleProjectWithSpringDMAndMavenPluginsFunctionalTest.java [hunk 2]
```diff
@@ -48,17 +50,27 @@ public void ensureProperPomGenerated() throws IOException, URISyntaxException, X
                 .forwardOutput()
                 .withPluginClasspath()
                 .build();
-        assertThat(buildResult.task(":install").getOutcome()).isEqualTo(TaskOutcome.SUCCESS);
+        assertThat(Objects.requireNonNull(buildResult.task(":install")).getOutcome()).isEqualTo(TaskOutcome.SUCCESS);
```
Comment (rnc) [inline manipulation/src/functTest/java/org/jboss/gm/manipulation/SimpleProjectWithSpringDMAndMavenPluginsFunctionalTest.java]: Is this due to warnings in IntelliJ ? - that the code there may throw `NullPointerException`? However, Objects.requireNonNull just throws NullPointerException so I'm not sure there is much benefit gained.

**Comment by rnc** [comment]
This is great! Just a couple of minor questions/comments...

File: manipulation/src/functTest/java/org/jboss/gm/manipulation/SimpleProjectWithSpringDMAndMavenPluginsFunctionalTest.java (hunk 2 above)
Comment (TomasHofman) [reply manipulation/src/functTest/java/org/jboss/gm/manipulation/SimpleProjectWithSpringDMAndMavenPluginsFunctionalTest.java]: Yes, this was just to get rid of Intellij warnings. I should probably use assertion instead, will fix.

**Comment by rnc** [comment]
Fantastic work :-)

**Pull request merged**